libsmtutil/CHCSmtLib2Interface.cpp

```diff
@@ -217,3 +217,21 @@ string CHCSmtLib2Interface::querySolver(string const& _input)
 	m_unhandledQueries.push_back(_input);
 	return "unknown\n";
 }
+
+std::string CHCSmtLib2Interface::getSMTLIBCode(Expression const& _expr)
+{
+	std::stringstream s;
+	s
+		<< "(set-option :timeout " + to_string(*m_queryTimeout) + ")" << std::endl
+		<< "(set-logic HORN)" << std::endl;
+
+	for (auto const& decl: m_smtlib2->userSorts() | ranges::views::values)
+		s << decl << std::endl;
```

Comment:
I think we also have a similar function that does that? Could also extract and re-use.